components/core/src/os/ffi/windows.rs

```diff
@@ -34,10 +35,10 @@ pub trait OsStrExt2 {
 impl OsStrExt3 for OsStr {
     fn from_bytes(b: &[u8]) -> &Self {
         use std::mem;
-        unsafe { mem::transmute(b) }
+        unsafe { mem::transmute(&*(b as *const [u8] as *const OsStr)) }
```

Comment:
What was Clippy's complaint about this?

I'm a little nervous about this change, given that it's in FFI Windows code using an incredibly unsafe Rust function.

Reply:
Yeah that's a fair concern.  I think I'll put it back how it was and explicitly allow it, with a note to revisit it at another time.